secp256k1-zkp-sys/src/lib.rs

```diff
@@ -0,0 +1,93 @@
+// secp256k1-zkp bindings
+// Written in 2019 by
+//   Jonas Nick
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! # Secp256k1-zkp
+//! Rust bindings for the secp256k1-zkp library.
+
+#![crate_type = "lib"]
+#![crate_type = "rlib"]
+#![crate_type = "dylib"]
+// Coding conventions
+#![deny(non_upper_case_globals)]
+#![deny(non_camel_case_types)]
+#![deny(non_snake_case)]
+#![deny(unused_mut)]
+#![warn(missing_docs)]
+#![cfg_attr(not(test), no_std)]
+#![cfg_attr(feature = "dev", allow(unstable_features))]
+#![cfg_attr(feature = "dev", feature(plugin))]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+
+#[cfg(any(test))]
+pub extern crate rand;
+
+#[cfg(test)]
+extern crate core;
+#[cfg(test)]
+extern crate secp256k1_zkp_dev;
+
+pub extern crate secp256k1;
+
+#[macro_use]
+mod macros;
+pub mod ffi;
+pub mod schnorrsig;
+mod types;
+
+/// The purpose of this rewriteable structure is to replace dynamic memory allocations in the
+/// libsecp256k1-zkp library. See its documentation for further information.
+pub struct ScratchSpace {
```

Comment:
I think we can impl both `Send` (no external references) and `Sync` (no interior mutability) on this.

Reply:
Perhaps I shouldn't do that because I don't understand the implications. Afaics `Send` is fine but `Sync` isn't (`Sync [...] answers the question: “if multiple threads were all trying to access this data, would it be safe?”` from https://manishearth.github.io/blog/2015/05/30/how-rust-achieves-thread-safety/).

Reply:
Yes, if multiple threads are trying to access the data it's fine. They can't mutate it because it's impossible for multiple threads to have ownership or `&mut` access simultaneously.

Reply:
done